engine/src/main/java/com/datatorrent/stram/StreamingAppMasterService.java

```diff
@@ -668,7 +671,10 @@ private void execute() throws YarnException, IOException
     int numReleasedContainers = 0;
     int nextRequestPriority = 0;
     ResourceRequestHandler resourceRequestor = new ResourceRequestHandler();
+    // Use override for container requestor in case of cloudera distribution,to handle host specific requests
+    ContainerRequestHandler containerRequestor = System.getenv().containsKey("CDH_HADOOP_BIN") ? new ContainerRequestHandlerCloudera() : new ContainerRequestHandler();
```

Comment:
Would it makes sense to combine ResourceRequestHandler and ContainerRequestHandler?